### ISSUE TITLE
some flag defaults show up as `[object Object]` in help text

### ISSUE THREAD

**Issue by cristiand391** [forcedotcom/cli#2539]
### Summary
some flags define a default value that the help text shows when using `--help`, but these are being rendered `[object Object]`.
Some examples:

![Screenshot 2023-10-27 at 11 11 33](https://github.com/forcedotcom/cli/assets/6853656/4cdd783b-cd8d-44f1-af64-48b83091d056)

![Screenshot 2023-10-27 at 11 13 24](https://github.com/forcedotcom/cli/assets/6853656/4a4b715d-bf92-484a-8b2a-a900bd401913)

### Steps To Reproduce

1) run `sf org create scratch -h` (or `--help`), see help text

### Expected result
default flag value is rendered correctly

### Actual result
`[object Object]`

### System Information

```json
{
  "architecture": "darwin-x64",
  "cliVersion": "@salesforce/cli/2.15.8",
  "nodeVersion": "node-v18.17.0",
  "osVersion": "Darwin 22.6.0",
  "rootPath": "/Users/cdominguez/.nvm/versions/node/v18.17.0/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@cristiand391/sf-plugin-api 0.0.2 (user)",
    "@cristiand391/sf-plugin-fzf-cmp 0.0.1 (user)",
    "@oclif/plugin-autocomplete 2.3.10 (core)",
    "@oclif/plugin-commands 3.0.4 (core)",
    "@oclif/plugin-help 6.0.4 (core)",
    "@oclif/plugin-not-found 3.0.2 (core)",
    "@oclif/plugin-plugins 3.9.4 (core)",
    "@oclif/plugin-search 1.0.4 (core)",
    "@oclif/plugin-update 4.1.3 (core)",
    "@oclif/plugin-version 2.0.3 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.2 (core)",
    "@oclif/plugin-which 3.0.6 (core)",
    "@salesforce/cli 2.15.8 (core)",
    "apex 2.3.20 (core)",
    "auth 2.8.23 (core)",
    "data 2.6.1 (core)",
    "deploy-retrieve 1.19.2 (core)",
    "dev 1.2.1 (user)",
    "info 2.6.51 (core)",
    "limits 2.3.41 (core)",
    "login 1.2.39 (core)",
    "marketplace 0.3.1 (core)",
    "org 2.11.5 (core)",
    "schema 2.3.31 (core)",
    "settings 1.4.36 (core)",
    "sobject 0.2.14 (core)",
    "source 2.10.45 (core)",
    "telemetry 2.3.8 (core)",
    "templates 55.5.17 (core)",
    "trust 2.6.22 (core)",
    "user 2.3.39 (core)"
  ]
}

```

**Comment by cristiand391** [comment]
fixed here: https://github.com/oclif/core/pull/844

should be available in sf >=v2.16

**Comment by jshackell-sfdc** [comment]
This issue is fixed in version [2.16.7 (Nov 8, 2023)](https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#2167-nov-8-2023-stable).
